src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -2425,6 +2425,25 @@ void PhaseIdealLoop::add_constraint(jlong stride_con, jlong scale_con, Node* off
     //     else /* scale > 0 and stride < 0 */
     //       I > (low_limit-(offset+1))/scale
     //   )
+    if (low_limit->get_long() == -max_jint) {
```

Comment:
There was similar code in both branches of the if ((stride_con^scale_con) >= 0) { but you're adding it back in only one of the branches. Is it not needed in the other one?

Reply:
I think I just missed on that one. In theory, we would just do more iterations in the pre-loop instead of the main-loop when underflowing and it will be fine. But I'm not sure if we eventually run into similar problems as with the other branch (I could not trigger that though). So, I reintroduced the code for this branch as well.